smelli/classes.py

```diff
@@ -313,28 +338,51 @@ def obstable(self, min_pull=0):
         due to correlations.
         """
         info = self._obstable_tree
-        info = self._obstable_filter_sort(info, min_pull=min_pull,
-                                          sortkey='pull', reverse=True)
+        subset = None
+        if sort_by == 'pull':
+            if ascending is None:
+                ascending = False
+            if min_val is not None:
+                min_val = max(min_pull, min_val)
+            else:
+                min_val = min_pull
+        elif min_pull != 0:
+            subset = lambda row: row['pull'] >= min_pull
+        if ascending is None:
+            ascending = True
+        info = self._obstable_filter_sort(info, sortkey=sort_by, ascending=ascending,
+                                          min_val=min_val, max_val=max_val,
+                                          subset=subset)
         # create DataFrame
         df = pd.DataFrame(info).T
         # remove inspire references, likelihood name, and observable name
-        del(df['inspire'])
-        del(df['lh_name'])
-        del(df['name'])
-        del(df['exp. PDF'])
+        if len(df) >0:
```

Comment:
When is the length 0? Can you add a comment please?

Reply:
Done in https://github.com/smelli/smelli/pull/1/commits/3f968936b1d471f26c63e697294e53110a6a4632.